dowhy/gcm/unit_change.py

```diff
@@ -0,0 +1,108 @@
+"""This module provides the APIs for attributing the change in the output value of a deterministic mechanism for a statistical unit.
+"""
+
+from abc import abstractmethod
+from typing import List, Optional
+
+import numpy as np
+import pandas as pd
+from sklearn.linear_model._base import LinearModel
+from sklearn.utils.validation import check_is_fitted
+
+from dowhy.gcm.fcms import PredictionModel
+from dowhy.gcm.ml.regression import SklearnRegressionModel
+from dowhy.gcm.shapley import ShapleyConfig, estimate_shapley_values
+
+
+class LinearPredictionModel:
```

Comment:
Can we make this a `Protocol`?

Reply:
Initially, I had it as a protocol. But then I dropped it because that would be inconsistent with existing `PredictionModel`.